src/clr-angular/data/datagrid/datagrid-row-detail.ts

```diff
@@ -67,14 +86,36 @@ export class ClrDatagridRowDetail<T = any> implements AfterContentInit, OnDestro
   public replacedRow = false;
 
   ngAfterContentInit() {
+    this.viewManager.detachAllViews(this._detailCells);
+    this.viewManager.insertAllViews(this._detailCells, this.assignRawOrders(), true);
     this.subscriptions.push(
       this.expand.replace.subscribe(replaceChange => {
         this.replacedRow = replaceChange;
+      }),
+      this.cells.changes.subscribe(() => {
+        this.viewManager.detachAllViews(this._detailCells);
+        this.viewManager.insertAllViews(this._detailCells, this.assignRawOrders(), true);
+      }),
+      // A subscription that listens for view reordering
+      this.columnReorderService.reorderRequested.subscribe(reorderRequest => {
+        const sourceView = this._detailCells.get(reorderRequest.sourceIndex);
+        this._detailCells.move(sourceView, reorderRequest.targetIndex);
       })
     );
   }
 
   ngOnDestroy() {
     this.subscriptions.forEach(sub => sub.unsubscribe());
   }
+
+  private assignRawOrders(): ClrDatagridCell[] {
```

Comment:
This is duplicated, if you wanted to put it into a shared service or not.

Reply:
This one is similar to one in `ClrDatagridRow`. `ClrDatagrid` has a method that has the same name as this one but the implementation is different. So basically only `ClrDatagridRow` and `ClrDatagridRowDetail` shares this method. And this doesn't really belong to any of the services we currently have and it's only one method so I decided to leave it in its individual component files.